src/acl/Checklist.h

```diff
@@ -209,6 +214,9 @@ class ACLChecklist
     std::stack<Breadcrumb> matchPath;
     /// the list of actions which must ignored during acl checks
     std::vector<Acl::Answer> bannedActions_;
+
+    /// the name of the last evaluated ACL (if any ACLs were evaluated)
+    std::optional<SBuf> lastCheckedName_;
```

Comment:
The `SBuf` object is essentially a smart pointer, so we should not use `std::optional` raw-pointers to them. Please try to replace this and other uses of `std::optional` with just `SBuf` variables (or `const SBuf &` in function parameters).

FWIW; Since all valid ACLs are named, the `SBuf::isEmpty()` method can be used to determine whether the name was set or not.

Reply:
> The `SBuf` object is essentially a smart pointer

I strongly disagree. In this discussion context, SBuf semantics is a container of characters, not a pointer. SBuf internals (that happen to use a pointer) are irrelevant.

We have abused empty SBuf to mean "no SBuf" in some contexts, mostly before we had access to std::optional, but there is no good reason to introduce the associated risks here and now.

> so we should not use `std::optional` raw-pointers to them. 

std::optional is not a raw pointer. At some level of abstraction, it can be viewed as a smart pointer. Overall, std::optional is much better than "some containers have special meaning" hacks.

> Since all valid ACLs are named, the `SBuf::isEmpty()` method can be used to determine whether the name was set or not.

Yes, the code will "work" if we assign an empty SBuf a special meaning of "unknown name", but there is no good reason for such error-prone hacks here.

> Please try to replace this and other uses of `std::optional` with just `SBuf`

I will not. Please withdraw this change request.